ir/src/symbol_table.rs

```diff
@@ -140,8 +123,17 @@ impl SymbolTable {
 
         let mut col_idx = 0;
         for trace_cols in trace {
-            let trace_column = TraceColumn::new(trace_segment, col_idx);
-            self.insert_symbol(trace_cols.name(), IdentifierType::TraceColumn(trace_column))?;
+            if trace_cols.size() == 1 {
+                let trace_column = TraceColumn::new(trace_segment, col_idx);
+                self.insert_symbol(trace_cols.name(), IdentifierType::TraceColumn(trace_column))?;
+            } else {
+                let trace_column_group =
+                    TraceColumnGroup::new(trace_segment, trace_cols.size() as usize, col_idx);
+                self.insert_symbol(
+                    trace_cols.name(),
+                    IdentifierType::TraceColumnGroup(trace_column_group),
+                )?;
+            }
```

Comment:
I'm not sure we need 2 different types here. What's your reasoning for not just having a single type like we did in the parser where for single columns the size of the group is 1?

Reply:
No good reason. Only that inserting access in graph is defined differently for symbol and vector so it seemed equivalent to me in terms of code readability/redundancy. If we go with defining one type, we will have to do checks in 2 places while inserting it in graph (one for symbol access and another for vector access). Please let me know if you prefer keeping one type here.

Reply:
Although maybe that check is redundant. We should only get vector access when size > 1.

Reply:
No good reason, I was mistaken. Fixed. Thanks.